UnnecessarySemicolonAfterTypeMemberDeclaration throws Exception on on Local Class

I have read check documentation: https://checkstyle.sourceforge.io/checks/coding/unnecessarysemicolonaftertypememberdeclaration.html#UnnecessarySemicolonAfterTypeMemberDeclaration
I have downloaded the latest checkstyle from: https://checkstyle.org/cmdline.html#Download_and_Run
I have executed the cli and showed it below, as cli describes the problem better than 1,000 words

```bash
➜  src javac SwitchScope.java 

➜  src cat SwitchScope.java 
public class SwitchScope {
    public static void main(String[] args) {
        switch (args.length) {
        case 0:
            final int k = 12;
            class Local {
                int j = k;
            }
        case 1:
        }
    }
}
                                                                                                           
➜  src cat config.xml 
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
    "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
    "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
  <property name="severity" value="error"/>
  <property name="fileExtensions" value="java, properties, xml"/>
  <module name="TreeWalker">
    <module name="UnnecessarySemicolonAfterTypeMemberDeclaration"/>
 </module>
</module>

➜  src java -jar checkstyle-10.12.1-all.jar -c config.xml SwitchScope.java 
Starting audit...
com.puppycrawl.tools.checkstyle.api.CheckstyleException: Exception was thrown while processing SwitchScope.java
        at com.puppycrawl.tools.checkstyle.Checker.processFiles(Checker.java:306)
        at com.puppycrawl.tools.checkstyle.Checker.process(Checker.java:223)
        at com.puppycrawl.tools.checkstyle.Main.runCheckstyle(Main.java:415)
        at com.puppycrawl.tools.checkstyle.Main.runCli(Main.java:338)
        at com.puppycrawl.tools.checkstyle.Main.execute(Main.java:195)
        at com.puppycrawl.tools.checkstyle.Main.main(Main.java:130)
Caused by: java.lang.NullPointerException: Cannot invoke "com.puppycrawl.tools.checkstyle.api.DetailAST.getType()" because "ast" is null
        at com.puppycrawl.tools.checkstyle.checks.coding.UnnecessarySemicolonAfterTypeMemberDeclarationCheck.isSemicolon(UnnecessarySemicolonAfterTypeMemberDeclarationCheck.java:236)
        at com.puppycrawl.tools.checkstyle.checks.coding.UnnecessarySemicolonAfterTypeMemberDeclarationCheck.checkTypeDefinition(UnnecessarySemicolonAfterTypeMemberDeclarationCheck.java:196)
        at com.puppycrawl.tools.checkstyle.checks.coding.UnnecessarySemicolonAfterTypeMemberDeclarationCheck.visitToken(UnnecessarySemicolonAfterTypeMemberDeclarationCheck.java:165)
        at com.puppycrawl.tools.checkstyle.TreeWalker.notifyVisit(TreeWalker.java:335)
        at com.puppycrawl.tools.checkstyle.TreeWalker.processIter(TreeWalker.java:406)
        at com.puppycrawl.tools.checkstyle.TreeWalker.walk(TreeWalker.java:273)
        at com.puppycrawl.tools.checkstyle.TreeWalker.processFiltered(TreeWalker.java:154)
        at com.puppycrawl.tools.checkstyle.api.AbstractFileSetCheck.process(AbstractFileSetCheck.java:98)
        at com.puppycrawl.tools.checkstyle.Checker.processFile(Checker.java:334)
        at com.puppycrawl.tools.checkstyle.Checker.processFiles(Checker.java:293)
        ... 5 more
Checkstyle ends with 1 errors.

```

---

**Describe what you expect in detail.**

I would expect this check not to throw a Null Pointer Exception on a local class declaration within a switch statement, even if this code is unusual.

First, I tried to reproduce an issue by creating a Test.java which actually replicates the issue and a config file. And, I got a nullpointer exception as seen below.

```
piyush@ Development % java -jar checkstyle-10.12.2-SNAPSHOT-all.jar -c config.xml Test.java
Starting audit...
com.puppycrawl.tools.checkstyle.api.CheckstyleException: Exception was thrown while processing Test.java
	at com.puppycrawl.tools.checkstyle.Checker.processFiles(Checker.java:306)
	at com.puppycrawl.tools.checkstyle.Checker.process(Checker.java:223)
	at com.puppycrawl.tools.checkstyle.Main.runCheckstyle(Main.java:415)
	at com.puppycrawl.tools.checkstyle.Main.runCli(Main.java:338)
	at com.puppycrawl.tools.checkstyle.Main.execute(Main.java:195)
	at com.puppycrawl.tools.checkstyle.Main.main(Main.java:130)
Caused by: java.lang.NullPointerException: Cannot invoke "com.puppycrawl.tools.checkstyle.api.DetailAST.getType()" because "ast" is null
	at com.puppycrawl.tools.checkstyle.checks.coding.UnnecessarySemicolonAfterTypeMemberDeclarationCheck.isSemicolon(UnnecessarySemicolonAfterTypeMemberDeclarationCheck.java:236)
	at com.puppycrawl.tools.checkstyle.checks.coding.UnnecessarySemicolonAfterTypeMemberDeclarationCheck.checkTypeDefinition(UnnecessarySemicolonAfterTypeMemberDeclarationCheck.java:196)
	at com.puppycrawl.tools.checkstyle.checks.coding.UnnecessarySemicolonAfterTypeMemberDeclarationCheck.visitToken(UnnecessarySemicolonAfterTypeMemberDeclarationCheck.java:165)
	at com.puppycrawl.tools.checkstyle.TreeWalker.notifyVisit(TreeWalker.java:335)
	at com.puppycrawl.tools.checkstyle.TreeWalker.processIter(TreeWalker.java:406)
	at com.puppycrawl.tools.checkstyle.TreeWalker.walk(TreeWalker.java:273)
	at com.puppycrawl.tools.checkstyle.TreeWalker.processFiltered(TreeWalker.java:154)
	at com.puppycrawl.tools.checkstyle.api.AbstractFileSetCheck.process(AbstractFileSetCheck.java:98)
	at com.puppycrawl.tools.checkstyle.Checker.processFile(Checker.java:334)
	at com.puppycrawl.tools.checkstyle.Checker.processFiles(Checker.java:293)
	... 5 more
Checkstyle ends with 1 errors.
```

After, i implemented the null-check, I get the following output without any nullpointer exception.

```
piyush@ Development % java -jar checkstyle-10.12.2-SNAPSHOT-all.jar -c config.xml Test.java
Starting audit...
Audit done.
```

Is it good enough? And, can i proceed with a draft PR or a PR?

--- a/config/projects-to-test/openjdk20-excluded.files
+++ b/config/projects-to-test/openjdk20-excluded.files
@@ -29,11 +29,6 @@
   <property name="fileNamePattern" value="[\\/]test[\\/]langtools[\\/]tools[\\/]javac[\\/]unicode[\\/]FirstChar2.java$"/>
 </module>
 
-<!-- until https://github.com/checkstyle/checkstyle/issues/13411 -->
-<module name="BeforeExecutionExclusionFileFilter">
-  <property name="fileNamePattern" value="[\\/]test[\\/]langtools[\\/]tools[\\/]javac[\\/]SwitchScope.java$"/>
-</module>
-
 <!-- non-compilable -->
 <module name="BeforeExecutionExclusionFileFilter">
   <property name="fileNamePattern" value="[\\/]test[\\/]langtools[\\/]tools[\\/]javac[\\/]diags[\\/]examples[\\/]ModuleDeclSbInModuleInfoJava.java$"/>

--- a/src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/UnnecessarySemicolonAfterTypeMemberDeclarationCheck.java
+++ b/src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/UnnecessarySemicolonAfterTypeMemberDeclarationCheck.java
@@ -233,6 +233,6 @@ public final class UnnecessarySemicolonAfterTypeMemberDeclarationCheck extends A
      * @return true if ast is semicolon, false otherwise
      */
     private static boolean isSemicolon(DetailAST ast) {
-        return ast.getType() == TokenTypes.SEMI;
+        return ast != null && ast.getType() == TokenTypes.SEMI;
     }
 }

--- a/src/test/java/com/puppycrawl/tools/checkstyle/checks/coding/UnnecessarySemicolonAfterTypeMemberDeclarationCheckTest.java
+++ b/src/test/java/com/puppycrawl/tools/checkstyle/checks/coding/UnnecessarySemicolonAfterTypeMemberDeclarationCheckTest.java
@@ -110,4 +110,13 @@ public class UnnecessarySemicolonAfterTypeMemberDeclarationCheckTest
             .that(check.getRequiredTokens())
             .isEqualTo(CommonUtil.EMPTY_INT_ARRAY);
     }
+
+    @Test
+    public void testIsSemicolonWithNullAst() throws Exception {
+        final String[] expected = {"24:32: " + getCheckMessage(MSG_SEMI)};
+
+        verifyWithInlineConfigParser(
+                getPath("InputUnnecessarySemicolonAfterTypeMemberDeclarationNullAst.java"),
+                expected);
+    }
 }

--- a/src/test/resources/com/puppycrawl/tools/checkstyle/checks/coding/unnecessarysemicolonaftertypememberdeclaration/InputUnnecessarySemicolonAfterTypeMemberDeclarationNullAst.java
+++ b/src/test/resources/com/puppycrawl/tools/checkstyle/checks/coding/unnecessarysemicolonaftertypememberdeclaration/InputUnnecessarySemicolonAfterTypeMemberDeclarationNullAst.java
@@ -1,0 +1,29 @@
+/*
+UnnecessarySemicolonAfterTypeMemberDeclaration
+tokens = (default)CLASS_DEF, INTERFACE_DEF, ENUM_DEF, ANNOTATION_DEF, VARIABLE_DEF, \
+         ANNOTATION_FIELD_DEF, STATIC_INIT, INSTANCE_INIT, CTOR_DEF, METHOD_DEF, \
+         ENUM_CONSTANT_DEF, COMPACT_CTOR_DEF, RECORD_DEF
+
+
+*/
+
+package com.puppycrawl.tools.checkstyle.checks.coding.unnecessarysemicolonaftertypememberdeclaration;
+
+public class InputUnnecessarySemicolonAfterTypeMemberDeclarationNullAst {
+    public static void main(String[] args) {
+        switch (args.length) {
+            case 0:
+                final int k = 12;
+                class Local1 {
+                    int j = k;
+                }
+                class Local2 {
+                    int j = k;
+                }; // ok
+                class Local3 {
+                    int l = 15;; //violation
+                }
+            case 1:
+        }
+    }
+}

